### PR TITLE
Button component

### PR DESCRIPTION
**Pull request by amyria3**
default / disabled state for primary / regular size
react component
basic story

### CODE REVIEW

**Comment by amyria3** [comment]
pls take a look

File: src/components/Button.tsx [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
 }:
 ButtonProps){
   const button = (
     <button
-
       type="button"
       className={
-        `text-center rounded-lg w-full h-12 text-primary text-dark-light ${(disabled)? 'bg-yellow-dimmed':'bg-yellow'}`
+        `text-center rounded-lg w-full text-primary
+        ${(primary)?'bg-yellow text-dark-light':'bg-dark-light text-white'}
```
Comment (dnmct) [inline src/components/Button.tsx:22]: use clsx for dynamically applying classes

File: src/components/Button.tsx [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+        ${(primary&&disabled)? 'bg-yellow-dimmed':''}
+        ${(default_height)? 'h-[49px]':'h-[38px]'}`
```
Comment (dnmct) [inline src/components/Button.tsx:24]: dont use fixed heights...
button size is determined by the padding

File: src/stories/Button.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { ComponentStory, ComponentMeta } from "@storybook/react";
```
Comment (dnmct) [inline src/stories/Button.stories.tsx:1]: those imports are deperecated use the storybook 7 syntax
https://storybook.js.org/docs/7.0/react/get-started/whats-a-story

**Review by dnmct — CHANGES_REQUESTED** [review]

File: src/components/Button.tsx (hunk 2 above)
Comment (amyria3) [reply src/components/Button.tsx:24]: but then the whole design make brake once you add a larger font size :)

File: src/stories/Button.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    //LINK my props to args.
+    primary: { control: "boolean" },
+    default_height: { control: "boolean" },
+  },
+};
+
+// const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />
+// //what is a template?
+
+// export const Primary = Template.bind({}) //what does .bind do
```
Comment (NoHara42) [inline src/stories/Button.stories.tsx:17]: Normally it'd be expected to remove large blocks of old template, but for the codecamp this is fine.

**Comment by amyria3** [comment]
All done :)

File: src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import clsx from "clsx";
+import React from "react";
+import "../index.css";
```
Comment (NoHara42) [inline src/components/Button.tsx:3]: index.css should be imported in your main.tsx, or app.tsx file... something higher up in the app hierarchy. 

then remove this here.

**Review by NoHara42 — CHANGES_REQUESTED** [review]
one change left before merge

**Comment by amyria3** [comment]
all done!

**Comment by NoHara42** [comment]
remember to reverse merge before claiming that you're done @amyria3 ;)

**Review by NoHara42 — APPROVED** [review]

**Pull request merged**